generator/metaline_builder.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include "generator/feature_builder.hpp"
+#include "generator/osm_element.hpp"
+
+#include <cstdlib>
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+namespace feature
+{
+class Segments;
+
+/// Merges road segments with similar name and ref values into groups called metalines.
+class MetalinesBuilder
+{
+public:
+  explicit MetalinesBuilder(std::string const & filePath) : m_filePath(filePath) {}
+  ~MetalinesBuilder() { Flush(); }
+  void operator()(OsmElement const & el, FeatureParams const & params);
```

Comment:
Please, insert blank lines between methods, short comment to operator and Flush would be helpfull.